docs/src/typescript.md

```diff
@@ -0,0 +1,34 @@
+## Manually compile tests with TypeScript
+
+Playwright Test supports TypeScript out the box. We automatically transform
+TypeScript code to javascript to run it.
+
+However if you find that the TypeScript code is not being transpiled correctly,
+you can perform your own TypeScript compilation before sending the tests to Playwright.
+
+First I add a `tsconfig.json` file inside my tests directory.
+```json
+{
+    "compilerOptions": {
+        "target": "ESNext",
+        "module": "commonjs",
+        "moduleResolution": "Node",
+        "sourceMap": true,
+        "outDir": "../tests-out",
+    }
+}
+```
+
+In my `package.json`, I have two scripts:
+```json
+{
+  "scripts": {
+    "pretest": "tsc --incremental -p tests/tsconfig.json",
+    "test": "playwright test -c tests-out"
+  }
+}
+```
+
+The `pretest` script runs typescript on the tests.  `test` will run the tests that have been generated to the `tests-out` directory.
+
+Then `npm run test` will build the tests and run them.
```

Comment:
hmm, how to use an existing tsconfig? Why does the playwright.config.ts need to be compiled? What is the -c parameter? Why is the config a directory?

Reply:
I think using an existing tsconfig is pretty straightforward once you read this doc.

There isn't anything here about playwright.config.ts, but it needs to be compiled the same as the tests because it might use syntax we dont support through babel. Complain to Dmitry about the `-c` command line api :).